packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -116,27 +115,31 @@ describe("BnsConnection", () => {
     profile: UserProfile,
     identity: PublicIdentity,
   ): Promise<void> {
-    const sendTx: SendTransaction = {
+    const sendTx = (await connection.withDefaultFee({
       kind: "bcp/send",
       creator: identity,
       recipient: await randomBnsAddress(),
       amount: defaultAmount,
-    };
+    })) as SendTransaction;
```

Comment:
Hmm, not so happy about all the necessary casting for the default use case.

Can't we use

```ts
const signed = await profile.signTransaction(await connection.withDefaultFee(sendTx), bnsCodec, nonce);
```

instead?

Reply:
See also other idea below

Reply:
I tried again to make types nicer and got this:

```ts
const sendTx = await connection.withDefaultFee<SendTransaction>({
      kind: "bcp/send",
      creator: identity,
      recipient: await randomBnsAddress(),
      amount: defaultAmount,
    });
```

We just enforce the generic type T of the function. Before we explicitly set `const signed: SendTransaction`. This does not force-cast the type, and will complain if the wrong string is used in kind. Does that work better for you? If so, I will update them all.

Reply:
I used this several places and I like it

Reply:
Yeah, cool, it is definitly better!